### PR TITLE
Separate prebuild PVC configuration

### PR DESCRIPTION
**Pull request by jenting**
## Description

Because we want the prebuild workspace volume snapshot to use a different VolumeSnapshotClass compared to the regular workspace volume snapshot.

For prebuild workspaces, the volume snapshot saves to GCP images.
For regular workspaces, the volume snapshot saves to GCP snapshots.

## Related Issue(s)

Partially #13980

## Release Notes

```release-note
None
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/ws-manager/pkg/manager/manager.go [hunk 1]
```diff
@@ -513,50 +520,6 @@ func (m *Manager) restoreVolumeSnapshotFromHandle(ctx context.Context, id, handl
 	return nil
 }
 
-func (m *Manager) DeleteVolumeSnapshot(ctx context.Context, req *api.DeleteVolumeSnapshotRequest) (res *api.DeleteVolumeSnapshotResponse, err error) {
```
Comment (sagor999) [inline components/ws-manager/pkg/manager/manager.go:516]: Why was this deleted? It is needed for GC to work.

**Review by sagor999 — COMMENTED** [review]
/hold

File: components/ws-manager/pkg/manager/manager.go (hunk 1 above)
Comment (sagor999) [reply components/ws-manager/pkg/manager/manager.go]: @jenting I reverted deletion of this function, I assume it was done by mistake. Reverted since you will be OOO until next week, and I would like to get this merged prior to that. LMK.

**Review by sagor999 — APPROVED** [review]
/unhold

File: components/ws-manager/pkg/manager/manager.go (hunk 1 above)
Comment (jenting) [reply components/ws-manager/pkg/manager/manager.go]: Yes, it was a mistake.

Comment (jenting) [reply components/ws-manager/pkg/manager/manager.go]: We need to pass the workspace type to the function `restore`.

Comment (sagor999) [reply components/ws-manager/pkg/manager/manager.go]: ah. hmm. that will require a bit more changes then, as I need to make webapp change to propagate it all the way from there (and hopefully that will not be too complicated). And it seems like that will require DB change as well to save what workspace type that volume snapshot belongs to.
So I will do that tomorrow morning.

Comment (jenting) [reply components/ws-manager/pkg/manager/manager.go]: If it's not a soft delete (i.e., it's a hard delete), I think we don't have to restore the VolumeSnapshot form handler and delete the VolumeSnapshot after that.

Therefore, we don't have to change the WebApp code and DB schema.

WDYT? @sagor999

Comment (sagor999) [reply components/ws-manager/pkg/manager/manager.go]: we have to, as otherwise we will not delete snapshot/image from gcp.

**Comment by sagor999** [comment]
Converted to draft. Working on a separate PR that adds workspace type to snapshot DB.

**Comment by sagor999** [comment]
Waiting for this [PR](https://github.com/gitpod-io/gitpod/pull/14209) to be reviewed and then will rebase this PR on top of it if necessary.

**Comment by sagor999** [comment]
@jenting I rebased on latest main and build is now working.
Do you have a related `ops` repo change for this?

**Comment by jenting** [comment]
> @jenting I rebased on latest main and build is now working. Do you have a related `ops` repo change for this?

Yes, https://github.com/gitpod-io/ops/pull/6195

File: install/installer/pkg/config/v1/config.md [hunk 2]
```diff
@@ -60,9 +60,12 @@ Config defines the v1 version structure of the gitpod config file
 |`workspace.templates.prebuild`||N|  ||
 |`workspace.templates.imagebuild`||N|  ||
 |`workspace.templates.regular`||N|  ||
-|`workspace.pvc.size`||Y|  |  Size is a size of persistent volume claim to use|
-|`workspace.pvc.storageClass`|string|N|  |  StorageClass is a storage class of persistent volume claim to use|
-|`workspace.pvc.snapshotClass`|string|N|  |  SnapshotClass is a snapshot class name that is used to create volume snapshot|
+|`workspace.prebuildPVC.size`||Y|  |  Size is a size of prebuild workspace persistent volume claim to use|
```
Comment (Pothulapati) [inline install/installer/pkg/config/v1/config.md:63]: Hmm. So, I was thinking if there was a better way to surface this config considering if there would be more things specific to prebuilds here. Without changing existing structure, How about `workspace.prebuild.PVC.*` ? 🤔

**Review by mads-hartmann — APPROVED** [review]
The changes to the files we currently own LGTM :shipit:

File: install/installer/pkg/config/v1/config.md (hunk 2 above)
Comment (jenting) [reply install/installer/pkg/config/v1/config.md:63]: @Pothulapati Thanks for your feedback.
If we choose the approach `workspace.prebuild.PVC.*`, we need to change the `workspace.pvc.*` to `workspace.regular.pvc.*`.

I think we could consider changing the structure once we want to support more custom settings between prebuild and regular workspace. WDYT?

**Review by Pothulapati — APPROVED** [review]
LGTM, on the installer side

**Comment by jenting** [comment]
@aledbf Could you review this one? Thank you.
